src/NServiceBus.AcceptanceTesting/AcceptanceTestingTransport/AcceptanceTestingTransportInfrastructure.cs

```diff
@@ -0,0 +1,146 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using DelayedDelivery;
+    using Performance.TimeToBeReceived;
+    using Routing;
+    using Settings;
+    using Transport;
+
+    class AcceptanceTestingTransportInfrastructure : TransportInfrastructure
+    {
+        public AcceptanceTestingTransportInfrastructure(SettingsHolder settings, bool nativePubSub, bool nativeDelayedDelivery)
+        {
+            this.nativePubSub = nativePubSub;
+            this.nativeDelayedDelivery = nativeDelayedDelivery;
+            this.settings = settings;
+
+            if (!settings.TryGet(StorageLocationKey, out storagePath))
+            {
+                var solutionRoot = FindSolutionRoot();
+                storagePath = Path.Combine(solutionRoot, ".learningtransport");
+            }
+
+            settings.SetDefault<MessageProcessingOptimizationExtensions.ConcurrencyLimit>(new MessageProcessingOptimizationExtensions.ConcurrencyLimit
+            {
+                MaxValue = 1
+            });
+
+            var errorQueueAddress = settings.ErrorQueueAddress();
+            PathChecker.ThrowForBadPath(errorQueueAddress, "ErrorQueueAddress");
+        }
+
+        public override IEnumerable<Type> DeliveryConstraints => nativeDelayedDelivery
+            ? new[]
+            {
+                typeof(DiscardIfNotReceivedBefore),
+                typeof(DelayDeliveryWith),
+                typeof(DoNotDeliverBefore)
+            }
+            : new[]
+            {
+                typeof(DiscardIfNotReceivedBefore)
+            };
+
+        public override TransportTransactionMode TransactionMode => TransportTransactionMode.SendsAtomicWithReceive;
+
+        public override OutboundRoutingPolicy OutboundRoutingPolicy => new OutboundRoutingPolicy(
+            OutboundRoutingType.Unicast, 
+            nativePubSub ? OutboundRoutingType.Multicast : OutboundRoutingType.Unicast, 
+            OutboundRoutingType.Unicast);
+
+        string FindSolutionRoot()
+        {
+            var directory = AppDomain.CurrentDomain.BaseDirectory;
+
+            while (true)
+            {
+                if (Directory.EnumerateFiles(directory).Any(file => file.EndsWith(".sln")))
+                {
+                    return directory;
+                }
+
+                var parent = Directory.GetParent(directory);
+
+                if (parent == null)
+                {
+                    // throw for now. if we discover there is an edge then we can fix it in a patch.
+                    throw new Exception("Couldn't find the solution directory for the learning transport. If the endpoint is outside the solution folder structure, make sure to specify a storage directory using the 'EndpointConfiguration.UseTransport<LearningTransport>().StorageDirectory()' API.");
```

Comment:
LearningTransport => AcceptanceTestTransport

Reply:
@andreasohlund  should be AcceptanceTestingTransport